js/about/preferences.js

```diff
@@ -317,6 +317,87 @@ class BitcoinDashboard extends ImmutableComponent {
   }
 }
 
+class PaymentHistory extends ImmutableComponent {
+  get ledgerData () {
+    return this.props.ledgerData
+  }
+
+  render () {
+    const transactions = this.props.ledgerData.get('transactions')
+
+    return <div id='paymentHistory'>
+      <table className='sort'>
+        <thead>
+          <tr>
+            <th className='sort-header' data-l10n-id='date' />
+            <th className='sort-header' data-l10n-id='totalAmount' />
+            <th className='sort-header' data-l10n-id='receiptLink' />
+          </tr>
+        </thead>
+        <tbody>
+        {
+          transactions.map(function (row) {
+            return <PaymentHistoryRow transaction={row} ledgerData={this.props.ledgerData} />
+          }.bind(this))
+        }
+        </tbody>
+      </table>
+    </div>
+  }
+}
+
+class PaymentHistoryRow extends ImmutableComponent {
+
+  get transaction () {
+    return this.props.transaction
+  }
+
+  get timestamp () {
+    return this.transaction.get('submissionStamp')
+  }
+
+  get formattedDate () {
+    return formattedDateFromTimestamp(this.timestamp)
+  }
+
+  get numericDateStr () {
+    var date = new Date(this.timestamp)
+
+    var monthNum = ('0' + (date.getMonth() + 1).toString())
+    monthNum = monthNum.slice(monthNum.length - 2, monthNum.length)
+
+    var dayNum = ('0' + (date.getDate().toString()))
+    dayNum = dayNum.slice(dayNum.length - 2, dayNum.length)
+
+    return monthNum + dayNum + date.getFullYear().toString()
+  }
+
+  get ledgerData () {
+    return this.props.ledgerData
+  }
+
+  get satoshis () {
+    return this.transaction.get('satoshis')
+  }
+
+  get totalAmount () {
+    var fiatAmount = this.satoshis * this.ledgerData.get('btcPrice') / 100000000
+    return fiatAmount.toFixed(2)
```

Comment:
I think this is available directly now, via `this.transaction.get('contribution').get('fiat').get('amount')`. see docs/state.md